src/xenstore.c

```diff
@@ -500,30 +508,76 @@ xenstore_destroy_usb(dominfo_t *domp, usbinfo_t *usbp)
 /**
  * Initialize the xenstore bits
  *
- * @return 0 on success, 1 on failure
+ * @return xenstorefd (>= 0) on success, -1 on failure
  */
 int
-xenstore_init(void)
+xenstore_init()
 {
+  unsigned int len;
+  char *domid_str;
+
   if (xs_handle == NULL) {
     xs_handle = xs_daemon_open();
   }
 
   if (xs_handle == NULL) {
     xd_log(LOG_ERR, "Failed to connect to xenstore");
-    return 1;
+    return -1;
   }
 
-  if (xs_dom0path == NULL) {
-    xs_dom0path = xs_get_domain_path(xs_handle, 0);
+  xs_backend_path = xs_get_domain_path(xs_handle, usb_backend_domid);
+
+  domid_str = xs_read(xs_handle, XBT_NULL, "domid", &len);
+  if (domid_str == NULL) {
+    xd_log(LOG_ERR, "Failed to read domid");
+    return -1;
   }
 
-  if (xs_dom0path == NULL) {
-    xd_log(LOG_ERR, "Could not get domain 0 path from XenStore");
-    return 1;
+  my_domid = atoi(domid_str);
```

Comment:
```atoi``` shouldn't blow up on you here but if ```domid_str``` is garbage or some large number, the ```atoi``` output could be misleading.  If you're only converting it from str -> int to log it, I'd feel more comfortable just using ```%s``` in the printf and using ```domid_str``` directly.

Reply:
We need the actual domid, so I can improve this.

Reply:
I still need to make this change in addition to the rebase.